include/boost/geometry/algorithms/detail/overlay/get_distance_measure.hpp

```diff
@@ -117,17 +116,32 @@ namespace detail
 // 0 (absolutely 0, not even an epsilon) means collinear. Like side,
 // a negative means that p is to the right of p1-p2. And a positive value
 // means that p is to the left of p1-p2.
-
 template <typename SegmentPoint, typename Point, typename Strategies>
 inline auto get_distance_measure(SegmentPoint const& p1, SegmentPoint const& p2, Point const& p,
                                  Strategies const&)
 {
-    return detail_dispatch::get_distance_measure
-            <
-                typename select_coordinate_type<SegmentPoint, Point>::type,
-                typename Strategies::cs_tag
-            >::apply(p1, p2, p);
+    using calc_t = typename select_coordinate_type<SegmentPoint, Point>::type;
+
+    // Verify equality, without using a tolerance
+    // (so don't use equals or equals_point_point)
+    // because it is about very tiny differences.
+    auto identical = [](const auto& point1, const auto& point2)
+    {
+        return geometry::get<0>(point1) == geometry::get<0>(point2)
+            && geometry::get<1>(point1) == geometry::get<1>(point2);
+    };
 
+    if (identical(p1, p) || identical(p2, p))
+    {
+        detail::distance_measure<calc_t> const result;
```

Comment:
is it safer and cleaner to set this explicitly to `0`?

Reply:
It's done in the constructor of `distance_measure`

Reply:
I know, I just comment that it is more clear to do explicit initialization so the reader will not have to see the declaration of the constructor, but that is my own preference. So I am resolving the comment :)

Reply:
Thanks. Agree it's more clear, but it's internal code and takes (a minimum amount) of time...